ocpmodels/trainers/energy_trainer.py

```diff
@@ -206,7 +216,11 @@ def predict(self, loader, results_file=None, disable_tqdm=False):
                 out["energy"] = self.normalizers["target"].denorm(
                     out["energy"]
                 )
-            predictions["id"].extend([str(i) for i in batch[0].sid.tolist()])
+            try:
+                predictions["id"].extend([str(i) for i in batch[0].sid.tolist()])
+            except:
+                #Not sure what we would want here or if something would work better in general
+                predictions["id"] == 'Unassigned'
```

Comment:
Let's avoid this try/except block, we also don't want `save_predictions` to be called when using the OCPCalculator. Can you restructure this so it follows the `ForcesTrainer` - https://github.com/Open-Catalyst-Project/ocp/blob/a267991ba8b895fe1a5cbe0194dd82aaf7ee0bcd/ocpmodels/trainers/forces_trainer.py#L297-L349?

Reply:
Also make sure to pull master into this and apply the `black` pre-commit hooks please.